community/kernel/src/main/java/org/neo4j/kernel/api/KernelTransaction.java

```diff
@@ -157,6 +157,20 @@
      */
     Type transactionType();
 
+    /**
+     * Return transaction id if it was already assigned.
```

Comment:
We should explain that it is only assigned in `TransactionEventHandler#afterCommit` and I'd like it to throw `IllegalStateException` if called from any other place, this will force user to use this correctly. Also then add that to throw javadoc.